applications/crossbar/src/modules_v1/cb_phone_numbers_v1.erl

```diff
@@ -461,6 +462,8 @@ validate_delete(Context) ->
                           cb_context:context().
 set_response({'ok', {'ok', Doc}}, _, Context) ->
     crossbar_util:response(Doc, Context);
+set_response({'ok', #knm_number{}=Number}, _, Context) ->
```

Comment:
Instead of matching here and making the records public, can you just create `knm_number:is_number/1` + do a `case`?

The rationale is to keep record inside its module in a kind of type-handling manner.